tests/api/api/test_schedules.py

```diff
@@ -80,203 +83,228 @@ def test_list_schedules(
+        [
+            "DELETE",
+            None,
+            http.HTTPStatus.NOT_FOUND.value,
+            {},
+        ],
         # we don't check if the project exists in update schedule, but rather query from the db and raise exception
-        # if schedule doesn't exists
-        {
-            "body": _create_schedule(),
-            "expected_status": http.HTTPStatus.NOT_FOUND.value,
-            "expected_body": {
+        # if schedule doesn't exist
+        [
+            "PUT",
+            tests.api.api.utils.compile_schedule(),
+            http.HTTPStatus.NOT_FOUND.value,
+            {
                 "detail": {
-                    "reason": f"MLRunNotFoundError('Schedule not found: project={project}, name={schedule_name}')"
+                    "reason": "MLRunNotFoundError('Schedule not found: project={project}, name={schedule_name}')"
                 }
             },
-        },
+        ],
         # updating schedule failed for unknown reason
-        {
-            "body": _create_schedule(),
-            "expected_status": http.HTTPStatus.NOT_FOUND.value,
-            "expected_body": {
+        [
+            "PUT",
+            tests.api.api.utils.compile_schedule(),
+            http.HTTPStatus.NOT_FOUND.value,
+            {
                 "detail": {
-                    "reason": f"MLRunNotFoundError('Schedule not found: project={project}, name={schedule_name}')"
+                    "reason": "MLRunNotFoundError('Schedule not found: project={project}, name={schedule_name}')"
                 }
             },
-        },
+        ],
         # project exists, expecting to create
-        {
-            "body": _create_schedule(),
-            "expected_status": http.HTTPStatus.OK.value,
-            "expected_body": {},
-        },
-    ]:
-        expected_status = test_case.get("expected_status")
-        expected_response = test_case.get("expected_body")
-        body = test_case.get("body")
+        ["PUT", tests.api.api.utils.compile_schedule(), http.HTTPStatus.OK.value, {}],
+    ],
+)
+@pytest.mark.asyncio
+async def test_redirection_from_worker_to_chief_schedule(
+    db: sqlalchemy.orm.Session,
+    async_client: httpx.AsyncClient,
+    aioresponses_mock: aioresponses_mock,
+    method: str,
+    body: dict,
+    expected_status: int,
+    expected_body: dict,
+):
+    project_name = "test-project"
+    schedule_name = "test_schedule"
+    endpoint, chief_mocked_url = _prepare_test_redirection_from_worker_to_chief(
+        project=project_name, endpoint_suffix=schedule_name
+    )
 
-        httpserver.expect_ordered_request(endpoint, method="PUT").respond_with_json(
-            expected_response, status=expected_status
+    # template the expected body
+    if "detail" in expected_body and "reason" in expected_body["detail"]:
+        expected_body["detail"]["reason"] = expected_body["detail"]["reason"].format(
+            project=project_name, schedule_name=schedule_name
```

Comment:
seems like you missed that for the update suite above, maybe make it a util so all will use it